storage/innobase/row/row0ins.cc

```diff
@@ -2619,6 +2619,25 @@ static uint64_t row_parse_int(const byte *data, size_t len,
   return 0;
 }
 
+/** Write the persistent auto increment value on clustered
+index root page.
+@param index 	clustered index
+@param auto_inc	next available AUTO_INCREMENT value
+@param mode	latch mode of the b-tree
+@param mtr	mini-transaction */
+static void row_ins_write_autoinc(const dict_index_t *index,
+				  uint64_t auto_inc,
+				  btr_latch_mode mode,
+				  mtr_t *mtr)
+{
+  if (auto_inc)
+  {
+    buf_block_t* root = mtr->at_savepoint(mode != BTR_MODIFY_ROOT_AND_LEAF);
```

Comment:
This function is missing `noexcept`. I think that the first two parameters had better be `mtr, auto_inc` so that no register shuffling will be needed for the first call.